GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -364,6 +366,8 @@ public virtual void OnIndexFileChange()
                 this.GitIndexProjection.InvalidateProjection();
                 this.InvalidateGitStatusCache();
             }
+
+            this.newlyCreatedFileAndFolderPaths.Clear();

```

Comment:
Since this change is only meant to help when some process is creating and deleting temporary files I decided to clear every index change to be safe and because `git status` would be some user initiated action but if they ran this while some process was creating and deleting files, some paths might not be removed.  So hopefully most of the time there won't be a bunch of files created, user runs some git command, and then the files are deleted.